src/PythonQtSignal.cpp

```diff
@@ -99,9 +99,10 @@ PythonQtSignalFunction_New(PythonQtSlotInfo *ml, PyObject *self, PyObject *modul
 /* Methods (the standard built-in methods, that is) */
 
 static void
-meth_dealloc(PythonQtSignalFunctionObject *m)
+meth_dealloc(PyObject *o)
```

Comment:
I don't see why this would change anything, but if it shuts up the checker...

Reply:
Probably, it is ([paragraph 7](https://en.cppreference.com/w/cpp/language/reinterpret_cast). And Python calls via pointer cast as `void (*)(PyObject*)`, thus it is an undefined behavior.

Reply:
That sounds right. I still don't think it will make a difference in practice, but you are right, the compiler is allowed to do anything in this case.